### PR TITLE
Allow using the Keystone v3 API

### PR DESCRIPTION
**Pull request by rhafer**
The crowbar-nova-set-availability-zone tool is still lacking support for
the v3 API. The same is true for the way how nova uses the neutronclient. We
hardcode the use of the v2 in both cases.

This depends on: https://github.com/crowbar/barclamp-keystone/pull/280

### CODE REVIEW

**Comment by dirkmueller** [comment]
+1

File: chef/cookbooks/nova/templates/default/nova.conf.erb [hunk 1]
```diff
@@ -540,7 +540,7 @@ ssl_key_file = <%= @ssl_key_file %>
 # max_header_line may need to be increased when using large
 # tokens (typically those generated by the Keystone v3 API
 # with big service catalogs). (integer value)
-#max_header_line=16384
+max_header_line = <%= node[:nova][:max_header_line] %>
```
Comment (vuntz) [inline chef/cookbooks/nova/templates/default/nova.conf.erb]: Don't we usually keep the commented defaults in the template?

Comment (tboerger) [inline chef/cookbooks/nova/templates/default/nova.conf.erb]: Usually yes

File: chef/data_bags/crowbar/migrate/nova/031_add_max_header_line.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+def upgrade ta, td, a, d
+  a['max_header_line'] = 16384
```
Comment (tboerger) [inline chef/data_bags/crowbar/migrate/nova/031_add_max_header_line.rb]: The correct thing should be ```a["max_header_line"] = ta["max_header_line"]```

**Pull request merged**